zerver/views/upload.py

```diff
@@ -71,5 +71,10 @@ def upload_file_backend(request, user_profile):
     if ((settings.MAX_FILE_UPLOAD_SIZE * 1024 * 1024) < user_file._get_size()):
         return json_error(_("File Upload is larger than allowed limit"))
 
+    if not isinstance(user_file.name, str):
+        # This check makes sure that the type of the file name is not
+        # Python 2 Unicode.
+        user_file.name = user_file.name.encode('ascii')
```

Comment:
Done. I have added the symptoms in the comments. I am not entirely sure why this is happening.